### PR TITLE
Fixed #29144 -- Made untranslated strings for territorial language variants use translations from the generic language variant.

### PR DESCRIPTION
**Pull request by patrys**
Added missing fallback tracking to DjangoTranslation.merge function to fix fallback handling for sparse territorial message catalogs.

Related ticket: https://code.djangoproject.com/ticket/29144

### CODE REVIEW

**Comment by patrys** [comment]
The isort failure looks like noise, this PR does not touch any imports.

**Comment by timgraham** [comment]
Please target the pull request to master. Even if the issue [qualifies for a backport](https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions) (most likely only if this issue is a regression), the committer will do that separately.

**Comment by patrys** [comment]
Retargetted for `master`

**Comment by patrys** [comment]
I've updated the docstring. I'll be happy to submit a separate PR to explain how precedence works with translations but I could not find the right place to mention in in the current docs. My gut feeling is that it could be a new heading right under "How Django discovers translations".

File: tests/i18n/tests.py [hunk 1]
```diff
@@ -1403,6 +1403,15 @@ def test_django_fallback(self):
         self.assertEqual(gettext('Date/time'), 'Datum/Zeit')
 
 
+@override_settings(INSTALLED_APPS=['i18n.fallback'])
+class TranslationFallbackI18NTests(ResolutionOrderI18NTests):
+
+    def test_sparse_territory_catalog(self):
```
[the remark this answers is not in the document]
Comment (patrys) [reply tests/i18n/tests.py]: Done 👍

**Comment by carltongibson** [comment]
@patrys Great. 

I've marked this RFC so Tim'll have a look. He'll have a better sense of what's best here. (If you think a new section, you can add it here in a new commit proactively.) 

Thanks for the effort!

**Comment by timgraham** [comment]
I amended the PR with some documentation. Could you check it?

**Comment by patrys** [comment]
Docs look good! :+1:

**Pull request merged**